### DISCUSSION TITLE
When/How do Rate limited jobs get processed from waiting queue when using the bounceBack = true option

### DISCUSSION THREAD

**Discussion by evilbuck** [OptimalBits/bull#1980] [Q&A]
When using the bounceBack option in the `RateLimiter` option, I understand they are sitting in the waiting queue. What is not clear to me is when or how those jobs in the waiting queue are picked up to be processed, or at all.  All the rate limited jobs sit in waiting. Is there some manual intervention that needs to happen?
```javascript

const queue = new Bull('test', {
  limiter: {
    max: 1,
    duration: 1000,
    bounceBack: true
  }
});

queue.process(async (job) => {
  console.log(`x:${x}`);
});

for(let x = 0; x < 10; x++ ) {
  queue.add({ x });
}

// yields
// # x:1
```

**Comment by manast** [comment]
The bounceBack option is AFAIK only useful for the manual processing pattern: https://github.com/OptimalBits/bull/blob/develop/PATTERNS.md#manually-fetching-jobs
If you use it in a normal queue it will keep bouncing until it is not rate limited anymore which probably will rise the redis instance CPU to 100% (and the process running bull as well).

**Reply by evilbuck** [reply]
got it. Thanks. fwiw, during my experimenting, I did not experience the redis cpu spiking. 
I was just looking for a way to let retry jobs process before rate limited jobs. This isn't it. Thanks again.
